addons/web_editor/controllers/main.py

```diff
@@ -323,23 +325,21 @@ def _attachment_create(self, name='', data=False, url=False, res_id=False, res_m
         else:
             raise UserError(_("You need to specify either data or url to create an attachment."))
 
-        # If the user is a portal and has access rights on the res_model (and
-        # on the res_id if it exists), we can create the attachment. Otherwise,
-        # one of the check will raise an error and the attachment will not be
-        # created
-        if request.env.user.has_group('base.group_portal'):
-            request.env[res_model].check_access_rights('write')
-            if res_id:
-                request.env[res_model].browse(res_id).check_access_rule('write')
-            attachment = request.env['ir.attachment'].sudo().create(attachment_data)
+        # Despite the user having no right to create an attachment, he can still
+        # create an image attachment through some flows
+        if (
+            not request.env.is_admin()
+            and IrAttachment._can_bypass_rights_on_media_dialog(**attachment_data)
+        ):
+            attachment = IrAttachment.sudo().create(attachment_data)
             # When portal users upload an attachment with the wysiwyg widget,
             # the access token is needed to use the image in the editor. If
             # the attachment is not public, the user won't be able to generate
             # the token, so we need to generate it using sudo
             if not attachment_data['public']:
                 attachment.sudo().generate_access_token()
```

Comment:
You will, through a later rpc, the code is in fact doing a first rpc the generate the attachment, then if necessary, it do another rpc to generate the access_token, which fail if you don't have the right.
In this particular condition going against the attachment rights, we need to generate ourselves so the later rpc doesn't fail (as it will not be triggered since not necessary to generate it).

See https://github.com/odoo/odoo/pull/82612#discussion_r899062243 and part of https://github.com/odoo/odoo/pull/82612#issuecomment-1175309757

Reply:
Ok I see :grimacing: